monitoring/parse_monitoring_log.py

```diff
@@ -0,0 +1,51 @@
+import sys
+
+
+def parse_task_log(log):
+    """ identify allocated and used memory and disk space within a cromwell task
+
+    :param [str] log: output generated by monitoring.sh script un via cromwell
+      on a google compute VM. Most often called 'monitoring.log', and found in
+      a task's cromwell directory within cromwell executions.
+    :return int memory_usage: maximum memory usage within task
+    :return int total_memory: memory allocated for task
+    :return bool robust_memory_estimate: True if there were 5 or more estimates of
+      the memory usage across the task's run.
+    :return int disk_usage: maximum disk usage within task
+    :return int total_disk: disk space allocated for task
+    :return bool robust_disk_estimate: True if there were 5 or more estimates of
+      the disk usage across the task's run.
+
+    """
+    total_memory = None
+    total_disk = None
+    memory_usage = []
+    disk_usage = []
+    for line in log:
+        if line.startswith('Total Memory:'):
+            total_memory = int(line.split()[-1])
+        elif line.startswith('Total Disk space:'):
+            total_disk = int(line.split()[-1])
+        elif line.startswith('* Memory usage (MB):'):
+            memory_usage.append(int(line.split()[-1]))
+        elif line.startswith('* Disk usage (KB):'):
+            disk_usage.append(int(line.split()[-1]))
+    robust_disk_estimate = len(disk_usage) >= 5
+    robust_memory_estimate = len(memory_usage) >= 5
+
+    return (
+        max(memory_usage), total_memory, robust_memory_estimate,
+        max(disk_usage), total_disk, robust_disk_estimate,
+    )
+
+if __name__ == "__main__":
+    if any(arg in {'-h', '--help', 'h'} for arg in sys.argv):
+        print('usage: python3 parse_monitoring_log.py')
+    for file_ in sys.argv[1:]:
+        with open(file_, 'r') as f:
+            print('file %s' % file_)
+            max_mem, tot_mem, _, max_disk, tot_disk, _ = parse_task_log(f.readlines())
```

Comment:
Does the monitoring script append a new resource utilization snapshot to a log file every 5 seconds? If so, you probably want to get rid of the `readlines` here, since I think that's going to try to read it all into memory.

Similarly, you might want to just store `max_memory_usage` and `max_disk_usage` rather than keeping the whole list around.

Reply:
Great suggestion; switched to iterators.